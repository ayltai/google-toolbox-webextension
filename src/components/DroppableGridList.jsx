import { Divider, GridList, makeStyles, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { SortableContainer, } from 'react-sortable-hoc';

import { DraggableButton, } from './DraggableButton';
import { ExpandablePanel } from './ExpandablePanel';

export const DroppableGridList = SortableContainer(({ services, configuration, }) => {
    const useStyles = makeStyles(() => ({
        divider : {
            width  : '100% !important',
            height : '1px !important',
        },
    }));

    const classes = useStyles();

    const [ expanded, setExpanded, ] = React.useState(false);

    const length = configuration.cols * configuration.rows;

    return (
        <ExpandablePanel onExpand={() => setExpanded(true)}>
            <GridList
                cellHeight={configuration.size + 24}
                cols={configuration.cols}>
                {services.slice(0, length).map((service, index) => (
                    <DraggableButton
                        key={service.id}
                        index={index}
                        service={service}
                        configuration={configuration} />
                ))}
                {expanded && (<Divider className={classes.divider} />)}
                {expanded && services.slice(length, services.length).map((service, index) => (
                    <DraggableButton
                        key={service.id}
                        index={index + length}
                        service={service}
                        configuration={configuration} />
                ))}
            </GridList>
        </ExpandablePanel>
    );
});

DroppableGridList.propTypes = {
    axis          : PropTypes.string,
    services      : PropTypes.arrayOf(PropTypes.object).isRequired,
    configuration : PropTypes.object.isRequired,
    onSortStart   : PropTypes.func,
    onSortMove    : PropTypes.func,
    onSortEnd     : PropTypes.func,
};
