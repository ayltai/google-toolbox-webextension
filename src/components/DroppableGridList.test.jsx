import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import browser from 'sinon-chrome/webextensions';

import { DroppableGridList } from './DroppableGridList';

window.browser = browser;

const services = [
    {
        id   : 'android-auto',
        name : 'Android Auto',
        url  : 'https://www.android.com/intl/en_us/auto/',
    },
    {
        id   : 'android-os',
        name : 'Android OS',
        url  : 'https://www.android.com/intl/en_us/',
    },
];

const configuration = {
    size : 48,
    cols : 4,
    rows : 2,
};

describe('<DroppableGridList />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <DroppableGridList
                services={services}
                configuration={configuration}
            />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <DroppableGridList
                services={services}
                configuration={configuration}
            />
        );
    });
});
