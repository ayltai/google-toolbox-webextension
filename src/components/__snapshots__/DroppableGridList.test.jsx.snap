// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DroppableGridList /> renders correctly 1`] = `
<div
  className="makeStyles-container-2"
>
  <ul
    className="MuiGridList-root"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-5 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Android Auto"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-6"
            >
              <img
                className="MuiAvatar-img"
                src="images/android-auto.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-7 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Android OS"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiAvatar-root MuiAvatar-square makeStyles-avatar-8"
            >
              <img
                className="MuiAvatar-img"
                src="images/android-os.png"
              />
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </ul>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      More
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
